Upgrade SDK (IdClient, Vaulting, CardFields, MCI)

nestor91sanchez opened the pull request.

Commit:
introducing new vaulting flow and implement CardFields

Commit:
Introducing CardFields

Commit:
added messaged when the payment method is not available

Commit:
updated vaulting scripts

suweed committed:
readme format

suweed committed:
Merge branch 'feature/upgrade-sdk' of bitbucket.org:qbo-digital/paypal into feature/upgrade-sdk

suweed committed:
readme format

suweed committed:
readme format

suweed committed:
readme format 2

suweed committed:
readme format 2

Commit:
fix styles and catch error on card field

Commit:
Merge remote-tracking branch 'upstream/feature/upgrade-sdk' into feature/upgrade-sdk

Commit:
added installment to buyer cost in admin

Commit:
added msi and mci

Commit:
added debug

Commit:
Merge branch 'master' into feature/upgrade-sdk

Commit:
fix debug

Commit:
hotfix

Commit:
change customerid preffix

Commit:
hotfix

Commit:
hotfix

Commit:
hotfix BA

suweed committed:
Merge branch 'master' into feature/upgrade-sdk

Commit:
improve code

Commit:
Merge remote-tracking branch 'upstream/feature/upgrade-sdk' into feature/upgrade-sdk

Commit:
added installment with cost to buyer to saved cards

suweed committed:
Logging & webhook error handling

Commit:
saved credit finance info to order detail

Commit:
Merge remote-tracking branch 'upstream/feature/upgrade-sdk' into feature/upgrade-sdk

suweed committed:
Add Payment Info in order email

suweed committed:
Payment info email fix

Commit:
load PayPal SDK immediately on first page load

Commit:
commented old paypalTokenAdapter and paypalSdkAdapter

Commit:
hotfix

Commit:
hotfix

Commit:
test

suweed committed:
Traduccion Comision

Commit:
fix smart buttons

Commit:
Merge remote-tracking branch 'upstream/feature/upgrade-sdk' into feature/upgrade-sdk

Commit:
added validation to show/hide remember card option

Commit:
fix installments validation

Commit:
formatted code

Commit:
testing

Commit:
Added Financing Options Ajax Validation

Commit:
hotfix

Commit:
fixed cardfield with new card and installments

qbo-tech commented on view/frontend/templates/paypal/sdk-loader.phtml:
```diff
@@ -0,0 +1,63 @@
+$isDebug = $block->isDebug();
+$nonce = $block->getNonce();
+
+$scriptUrl = $paypalUrl . ($enableVault && $enableAcdc ? '&vault=true' : '');
```
Si no esta habilitada ninguna de las opciones de pago del plugin (El flujo estandar, BCDC, ACDC, o RT) no se debe cargar el SDK.

qbo-tech commented on Plugin/AddPaypalInstallments.php:
```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Paypal\CommercePlatform\Plugin;
```
Redundante, remover.
Ya se hace en Block/Info.php

Commit:
removed customerId preffix

Commit:
removed plugin and change error message

Commit:
validate if paypal is active

qbo-tech committed:
Credit Memo Refund Fix

Commit:
Refreshes SPB buttons on grand total change

Commit:
Merge branch 'master' into feature/upgrade-sdk

Commit:
Merge branch 'master' into feature/upgrade-sdk

Commit:
Merge branch 'feature/upgrade-sdk'

Commit:
Merge branch 'master' into feature/upgrade-sdk

nestor91sanchez committed:
Validates PayPal payment amount matches order total

nestor91sanchez committed:
Implements amount mismatch validation and improves webhook logging

- Adds a configuration to halt the capture flow if the PayPal order amount differs from the Magento grand total.
- Enhances webhook debugging with detailed logging and a signature verification bypass for sandbox environments.
- Ensures order IDs are reserved before constructing API request bodies for financing and standard orders.

nestor91sanchez committed:
Fixes HTML nesting and indentation in the PayPal Advanced form template

Corrects the template structure by properly closing the label container and de-indenting the payment method content block to ensure proper layout and readability.
